### PR TITLE
feat(*): add link selector

### PR DESCRIPTION
**Pull request by fulopdaniel**
**guideline-blocks-settings:**
Create the LinkInput component that is the wrapper of a text input and the internal link chooser button. This component will be used inside the RTE modals, and inside the settings sidebar. Also some refactoring, moving the link chooser outside the RTE folder.

**sidebar-settings:**
Introduce the new link block. Similar to the link chooser block but with the new link chooser.

### CODE REVIEW

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                <TextInput
+                    id="url"
+                    value={url}
+                    clearable={clearable}
+                    onChange={onUrlChange}
+                    placeholder={placeholder ?? 'https://example.com'}
+                    focusOnMount
+                />
+            </FormControl>
+            {!isUrlValid && <div className="tw-text-red-65 tw-mt-1 tw-text-s">Please enter a valid URL.</div>}
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx:61]: `tw-text-red-65` izn't a valid token

File: packages/guideline-blocks-settings/src/components/Link/types.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+export type InitiallyExpandedItems = {
+    documentId: number | undefined;
+    pageId: number | undefined;
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/types.ts:5]:
```suggestion
    documentId?: number;
    pageId?: number;
```

File: packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx [hunk 3]
```diff
@@ -9,10 +9,11 @@ import { DocumentLinks } from './DocumentLinks';
 type LinkSelectorProps = {
     appBridge: AppBridgeBlock;
     url: string;
-    onUrlChange: (value: string) => void;
+    onUrlChange?: (value: string) => void;
+    buttonSize?: ButtonSize;
 };
 
-export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps): ReactElement => {
+export const LinkSelector = ({ appBridge, url, onUrlChange, buttonSize }: LinkSelectorProps): ReactElement => {
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx:16]:
```suggestion
export const LinkSelector = ({ appBridge, url, onUrlChange, buttonSize = ButtonSize.Medium }: LinkSelectorProps): ReactElement => {
```

File: packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx [hunk 4]
```diff
@@ -33,15 +34,17 @@ export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps)
...
+        }
         closeLinkTree();
     };
 
     return (
         <div data-test-id="internal-link-selector" onKeyDown={onPressEnter}>
             <Button
                 icon={<IconLink />}
-                size={ButtonSize.Medium}
+                size={buttonSize ?? ButtonSize.Medium}
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx:47]:
```suggestion
                size={buttonSize}
```

File: packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx [hunk 5]
```diff
@@ -33,15 +34,17 @@ export const LinkSelector = ({ appBridge, url, onUrlChange }: LinkSelectorProps)
     }, [url, selectedUrl]);
 
     const saveLink = () => {
-        onUrlChange(selectedUrl);
+        if (onUrlChange) {
+            onUrlChange(selectedUrl);
+        }
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx:39]:
```suggestion
        onUrlChange?.(selectedUrl);
```

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+type LinkInputProps = {
+    url?: string;
+    info?: string;
+    label?: string;
+    required?: boolean;
+    newTab?: CheckboxState;
+    openInNewTab?: boolean;
+    onUrlChange?: (value: string) => void;
+    onToggleTab?: (checked: boolean) => void;
+    isValidUrlOrEmpty?: (url: string) => boolean | undefined;
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx:16]:
```suggestion
    isValidUrlOrEmpty?: (url: string) => boolean;
```

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.spec.ct.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+        cy.get(BUTTON_ID).should('not.exist');
+    });
+
+    it('renders * if input is required', () => {
+        const [LinkInputWithStubs] = withAppBridgeBlockStubs(LinkInput, {});
+        mount(<LinkInputWithStubs required label="Link" />);
+
+        cy.get(INPUT_LABEL_CONTAINER_ID).contains('*');
+    });
+    it('renders custom buttonsize', () => {
```
Comment (SamuelAlev) [inline packages/guideline-blocks-settings/src/components/Link/LinkInput.spec.ct.tsx:135]:
```suggestion
    });

    it('renders custom buttonsize', () => {
```

File: packages/guideline-blocks-settings/src/index.ts [hunk 8]
```diff
@@ -124,9 +126,9 @@ export const defineSettings = <
     T extends
         | BlockSettingsStructureExport
         | (() => Promise<BlockSettingsStructureExport>)
-        | (() => BlockSettingsStructureExport),
+        | (() => BlockSettingsStructureExport)
 >(
-    settingsStructure: T,
+    settingsStructure: T
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/index.ts:131]: image
 
 Seems like they reverted to 2.x.x for a while.

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx (hunk 1 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx:61]: image
image
 
 It seems like it is.

File: packages/sidebar-settings/src/blocks/link.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * The placeholder text for the link chooser.
+     */
+    placeholder?: string;
+
+    /**
+     * Whether the link should be opened in a new tab or not.
+     */
+    openInNewTab?: boolean;
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/sidebar-settings/src/blocks/link.ts:20]: Added 👍

File: packages/sidebar-settings/src/blocks/link.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+    /**
+     * Whether the link chooser can be cleared or not.
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/sidebar-settings/src/blocks/link.ts:23]: Added 👍

File: packages/guideline-blocks-settings/src/components/Link/types.ts (hunk 2 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Link/types.ts]: Fixed

File: packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx (hunk 4 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Link/LinkSelector/LinkSelector.tsx]: Fixed

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.spec.ct.tsx (hunk 7 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Link/LinkInput.spec.ct.tsx]: Fixed

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx (hunk 1 above)
Comment (SamuelAlev) [reply packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx:62]: Sorry I meant part of these tokens: https://beta-fondue-components.frontify.com/?path=/story/tokens--alias-tokens

File: packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/FloatingLink/InsertLinkModal/useInsertModal.ts [hunk 11]
```diff
@@ -119,7 +119,7 @@ export const useInsertModal = () => {
         {
             enableOnFormTags: ['INPUT'],
         },
-        [],
+        []
```
Comment (ragi96) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/plugins/LinkPlugin/FloatingLink/InsertLinkModal/useInsertModal.ts:122]: also prettier

File: packages/guideline-blocks-settings/src/components/RichTextEditor/serializer/serializeToHtml.ts [hunk 12]
```diff
@@ -9,7 +9,7 @@ export const serializeRawToHtmlAsync = async (
     raw: string,
```
Comment (ragi96) [inline packages/guideline-blocks-settings/src/components/RichTextEditor/serializer/serializeToHtml.ts:9]: There shouldn't be any changes in here, prettier ;)

File: packages/guideline-blocks-settings/src/index.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/index.ts:131]: Had to restart VS code now it works 🤦

File: packages/guideline-blocks-settings/src/index.ts [hunk 13]
```diff
@@ -46,6 +47,7 @@ export type FontInputBlock = FontInputBlockSidebarSettings<AppBridgeBlock>;
 export type InputBlock = InputBlockSidebarSettings<AppBridgeBlock>;
 export type LegacyAssetInputBlock = LegacyAssetInputBlockSidebarSettings<AppBridgeBlock>;
 export type LinkChooserBlock = LinkChooserBlockSidebarSettings<AppBridgeBlock>;
+export type LinkBlock = LinkBlockSidebarSettings<AppBridgeBlock>;
```
[the remark this answers is not in the document]
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/index.ts]: Fixed!

File: packages/guideline-blocks-settings/src/components/RichTextEditor/serializer/serializeToHtml.ts (hunk 12 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/RichTextEditor/serializer/serializeToHtml.ts]: Fixed, and all the other prettier issues as well 😇

File: packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx (hunk 1 above)
Comment (fulopdaniel) [reply packages/guideline-blocks-settings/src/components/Link/LinkInput.tsx]: Ah ok, fixed

**Review by ragi96 — COMMENTED** [review]
This is not really related to your changes, but i will add it here.

The ordering in the linkChooser on page level isn't correct:

Link Chooser ordering:
Screenshot 2023-08-10 at 10 37 49

Nav Manager & Sidebar:
Screenshot 2023-08-10 at 10 40 50

**Review by ragi96 — APPROVED** [review]
Thanks, looks good now 👍

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**